target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8072-hh500v.dts

```diff
@@ -0,0 +1,421 @@
+		serial0 = &blsp1_uart5;
+		serial1 = &blsp1_uart3;
+		ethernet3 = &dp4;
+		ethernet5 = &dp6_syn;
+	};
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+		bootargs-append = " root=/dev/ubiblock0_1";
+	};
+
+	modem {
+		compatible = "gpio-export";
+		pinctrl-names = "default";
+		pinctrl-0 = <&modem_pins>;
+
+		/* output low = download mode */
+		modem_forcedl {
+			gpio-export,name = "modem_forcedl";
+			gpio-export,output = <1>;
+			gpios = <&tlmm 46 GPIO_ACTIVE_HIGH>;
+		};
+
+		/* output high = power on, output low = no power on after reset */
+		modem-power {
+			gpio-export,name = "modem_power";
+			gpio-export,output = <1>;
+			gpios = <&tlmm 30 GPIO_ACTIVE_HIGH>;
+		};
+
+		/*
+		 * output low 1s pulse = reset
+		 * sometimes crashes kernel if mhi_pci_generic is active
+		 */
+		modem-reset {
+			gpio-export,name = "modem_reset";
+			gpio-export,output = <1>;
+			gpios = <&tlmm 29 GPIO_ACTIVE_HIGH>;
+		};
+	};
+
+	gpio-poweroff {
+		compatible = "gpio-poweroff";
+		gpios = <&tlmm 20 GPIO_ACTIVE_HIGH>;
+	};
+
+	keys {
+		compatible = "gpio-keys";
+		pinctrl-0 = <&button_pins>;
+		pinctrl-names = "default";
+
+		reset {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&tlmm 47 GPIO_ACTIVE_LOW>;
+		};
+
+		wps {
+			label = "wps";
+			linux,code = <KEY_WPS_BUTTON>;
+			gpios = <&tlmm 64 GPIO_ACTIVE_LOW>;
+		};
+
+		power {
+			label = "power";
+			gpios = <&tlmm 62 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_POWER>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		pinctrl-names = "default";
+		pinctrl-0 = <&led_pins>;
+
+		4g_green {
+			label = "green:4g";
+			color = <LED_COLOR_ID_GREEN>;
+			gpios = <&aw9523 0 GPIO_ACTIVE_LOW>;
+		};
+
+		4g_blue {
+			label = "blue:4g";
+			color = <LED_COLOR_ID_BLUE>;
+			gpios = <&aw9523 1 GPIO_ACTIVE_LOW>;
+		};
+
+		wifi_blue {
+			label = "blue:wifi";
+			color = <LED_COLOR_ID_BLUE>;
+			gpios = <&aw9523 3 GPIO_ACTIVE_LOW>;
+		};
+
+		led_power_blue: power_blue {
+			label = "blue:power";
+			color = <LED_COLOR_ID_BLUE>;
+			gpios = <&aw9523 5 GPIO_ACTIVE_LOW>;
+		};
+
+		led_power_red: power_red {
+			label = "red:power";
+			color = <LED_COLOR_ID_RED>;
+			gpios = <&aw9523 6 GPIO_ACTIVE_LOW>;
+		};
+
+		5g_red {
+			label = "red:5g";
+			color = <LED_COLOR_ID_RED>;
+			gpios = <&aw9523 8 GPIO_ACTIVE_LOW>;
+		};
+
+		5g_green {
+			label = "green:5g";
+			color = <LED_COLOR_ID_GREEN>;
+			gpios = <&aw9523 9 GPIO_ACTIVE_LOW>;
+		};
+
+		5g_blue {
+			label = "blue:5g";
+			color = <LED_COLOR_ID_BLUE>;
+			gpios = <&aw9523 10 GPIO_ACTIVE_LOW>;
+		};
+
+		4g_red {
+			label = "red:4g";
+			color = <LED_COLOR_ID_RED>;
+			gpios = <&aw9523 11 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+&tlmm {
+	mdio_pins: mdio-pins {
+		mdc {
+			pins = "gpio68";
+			function = "mdc";
+			drive-strength = <8>;
+			bias-pull-up;
+		};
+
+		mdio {
+			pins = "gpio69";
+			function = "mdio";
+			drive-strength = <8>;
+			bias-pull-up;
+		};
+	};
+
+	button_pins: button-pins {
+		power_button {
```

Comment:
Lets get rid of these QCA style subnodes, no point in them since you are applying literally the same config for all of these pins